ruby/ql/lib/codeql/ruby/frameworks/Sinatra.qll

```diff
@@ -179,6 +179,12 @@ module Sinatra {
     }
   }
 
+  bindingset[local]
+  pragma[inline_late]
+  private predicate isPairKey(string local) {
```

Comment:
You still get a lot of fanout at the end of your pipeline, which looks like it could be avoided if this predicate was `nomagic`ed (it very well might still need `bindingset`+`inline_late` in which case it would need a wrapper).

Reply:
Yeah, I thought about doing that, but decided not to given the relatively small tuple counts.